src/shared/user-record.c

```diff
@@ -1747,7 +1747,7 @@ const char *user_record_shell(UserRecord *h) {
                 return "/bin/sh";
 
         if (user_record_disposition(h) == USER_REGULAR)
-                return "/bin/bash";
+                return access("/bin/bash", X_OK) ? "/bin/sh" : "/bin/bash";
```

Comment:
This is called to answer D-Bus messages, so it could be running under a different filesystem, so a check like this might not be valid